rust_src/src/fonts.rs

```diff
@@ -0,0 +1,64 @@
+use remacs_macros::lisp_fn;
+use remacs_sys::{PseudovecType, Qfont_spec, Qfont_entity, Qfont_object, Qsymbolp,
+                 wrong_type_argument};
+use lisp::{LispObject, Qnil};
+
+// See font_property_index in font.h for details.
+#[allow(non_camel_case_types, dead_code)]
+#[repr(C)]
+enum font_property_index {
+    FONT_TYPE_INDEX,
+    FONT_FOUNDRY_INDEX,
+    FONT_FAMILY_INDEX,
+    FONT_ADSTYLE_INDEX,
+    FONT_REGISTRY_INDEX,
+    FONT_WEIGHT_INDEX,
+    FONT_SLANT_INDEX,
+    FONT_WIDTH_INDEX,
+    FONT_SIZE_INDEX,
+    FONT_DPI_INDEX,
+    FONT_SPACING_INDEX,
+    FONT_AVGWIDTH_INDEX,
+    FONT_EXTRA_INDEX,
+    // In C, we have FONT_SPEC_MAX, FONT_OBJLIST_INDEX = FONT_SPEC_MAX here.
+    FONT_OBJLIST_INDEX,
+    // In C, we have FONT_ENTITY_MAX, FONT_NAME_INDEX = FONT_ENTITY_MAX here.
+    FONT_NAME_INDEX,
+    FONT_FULLNAME_INDEX,
+    FONT_FILE_INDEX,
+    // In C, we have FONT_OBJECT_MAX here.
+}
+
+const FONT_SPEC_MAX: i32 = font_property_index::FONT_OBJLIST_INDEX as i32;
+const FONT_ENTITY_MAX: i32 = font_property_index::FONT_NAME_INDEX as i32;
+const FONT_OBJECT_MAX: i32 = (font_property_index::FONT_FILE_INDEX as i32) + 1;
+
+/// Return t if OBJECT is a font-spec, font-entity, or font-object.
+/// Return nil otherwise.
+/// Optional 2nd argument EXTRA-TYPE, if non-nil, specifies to check
+/// which kind of font it is.  It must be one of `font-spec', `font-entity',
+/// `font-object'.
+#[lisp_fn(min = "1")]
+pub fn fontp(object: LispObject, extra_type: LispObject) -> LispObject {
+    object.as_vectorlike().map_or(Qnil, |v| {
+        if v.is_pseudovector(PseudovecType::PVEC_FONT) {
+            if extra_type.eq(Qnil) {
+                LispObject::constant_t()
+            } else if extra_type.eq(LispObject::from_raw(unsafe { Qfont_spec })) {
```

Comment:
Let me clarify:

Accessing these symbols the way you are doing is fine, but we have to use unsafe blocks in Lisp binding code. If we have another lisp binding that needs to do similar logic, we will need to add more unsafe blocks. In general, I think a Rust code base is at it's best when it minimizes unsafe code blocks. 

We can improve this situation by adding a safe wrapper around unsafe code (since the code here is "unsafe" due to its interaction with C). For example something like:

```
impl LispObject {
    pub fn is_font_spec(&self) -> bool {
        self.eq(unsafe { /* ... */ });  
    }
}
```